make/autoconf/jdk-options.m4

```diff
@@ -415,19 +415,29 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
       CHECK_AVAILABLE: [
         AC_MSG_CHECKING([if AddressSanitizer (asan) is available])
         if test "x$TOOLCHAIN_TYPE" = "xgcc" ||
-            test "x$TOOLCHAIN_TYPE" = "xclang"; then
+           test "x$TOOLCHAIN_TYPE" = "xclang" ||
+           test "x$TOOLCHAIN_TYPE" = "xmicrosoft"; then
           AC_MSG_RESULT([yes])
         else
           AC_MSG_RESULT([no])
           AVAILABLE=false
         fi
       ],
       IF_ENABLED: [
-        # ASan is simply incompatible with gcc -Wstringop-truncation. See
-        # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85650
-        # It's harmless to be suppressed in clang as well.
-        ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
-        ASAN_LDFLAGS="-fsanitize=address"
+        ASAN_CFLAGS="-fsanitize=address"
+        ASAN_LDFLAGS=""
+        if test "x$TOOLCHAIN_TYPE" = "xgcc" ||
+           test "x$TOOLCHAIN_TYPE" = "xclang"; then
+          # ASan is simply incompatible with gcc -Wstringop-truncation. See
+          # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85650
+          # It's harmless to be suppressed in clang as well.
+          ASAN_CFLAGS="$ASAN_CFLAGS -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common"
+          ASAN_LDFLAGS="$ASAN_LDFLAGS -fsanitize=address"
+        elif test "x$TOOLCHAIN_TYPE" = "xmicrosoft"; then
+          # MSVC produces a warning if you pass -fsanitize=address to the linker.
+          ASAN_CFLAGS="$ASAN_CFLAGS -Oy-"
```

Comment:
What is the purpose of `/Oy-`? Can you explain it in a short comment?

Reply:
Added a comment.